File: historyscan-back/src/test/java/io/sch/historyscan/domain/contexts/analysis/network/NetworkLinkTest.java
package io.sch.historyscan.domain.contexts.analysis.network;

import org.junit.jupiter.api.Test;

import java.util.List;
import java.util.stream.Stream;

import static org.assertj.core.api.Assertions.assertThat;

class NetworkLinkTest {

    @Test
    void should_sort_links() {
        final NetworkLink link1 = new NetworkLink("theglobalproject/src/main/java/io/mycompany/theproject/domain/boundedcontexts/featureB/FinallyLoading.java", 15);
        final NetworkLink link2 = new NetworkLink("theglobalproject/src/main/java/io/mycompany/theproject/domain/boundedcontexts/featureA/refactored/Extensions.java", 14);
        final NetworkLink link3 = new NetworkLink("theglobalproject/src/main/java/io/mycompany/theproject/domain/boundedcontexts/featureB/refactored/Loading.java", 14);
        final NetworkLink link4 = new NetworkLink("b", 15);
        final NetworkLink link5 = new NetworkLink("z", 1);
        var networkLinks = Stream.of(link5, link4, link3, link1, link2).sorted().toList();
        assertThat(networkLinks).isEqualTo(List.of(link4, link1, link2, link3, link5));
    }
}

File: historyscan-back/src/test/kotlin/io/sch/historyscan/domain/contexts/analysis/network/NetworkLinkTest.kt
package io.sch.historyscan.domain.contexts.analysis.network

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import java.util.List
import java.util.stream.Stream

internal class NetworkLinkTest {
    @Test
    fun should_sort_links() {
        val link1 = NetworkLink(
            "theglobalproject/src/main/java/io/mycompany/theproject/domain/boundedcontexts/featureB/FinallyLoading.java",
            15
        )
        val link2 = NetworkLink(
            "theglobalproject/src/main/java/io/mycompany/theproject/domain/boundedcontexts/featureA/refactored/Extensions.java",
            14
        )
        val link3 = NetworkLink(
            "theglobalproject/src/main/java/io/mycompany/theproject/domain/boundedcontexts/featureB/refactored/Loading.java",
            14
        )
        val link4 = NetworkLink("b", 15)
        val link5 = NetworkLink("z", 1)
        val networkLinks = Stream.of(link5, link4, link3, link1, link2).sorted().toList()
        Assertions.assertThat(networkLinks).isEqualTo(List.of(link4, link1, link2, link3, link5))
    }
}